Automattic/woocommerce-payments
# Prompt if business location is not among supported countries when onboarding merchants

Fixes #694 

#### Changes proposed in this Pull Request

This PR introduces a modal window, which appears during WooCommerce Payments set up if merchant business country (identified from the address specified in WooCommerce settings) is not supported. When the country is supported, the dialog is not shown and the merchant follows the regular Set up workflow.

#### Testing instructions

* Under 'WooCommerce -> Settings' change the business country to one we are not supporting yet ([see supported countries list](https://github.com/Automattic/woocommerce-payments/blob/add/694-onboarding-store-location-check/includes/class-wc-payments-utils.php#L178))
* Enable 'Force onboarding' under Dev-tools and navigate to WooCommerce Payments
* The setup screen expected to appear, click 'Set up' button
* When clicked, a modal window should appear listing the supported countries and prompting that merchant country is not supported
* Click 'Cancel': the dialog should disappear and UI resume to the original state
* Trigger the dialog again
* Click 'Continue': you should be redirected to Stripe (please note the Cancel and Continue buttons are getting disabled)

After reverting settings changes, the Setup page expected to Send you to Stripe without the modal.

-------------------
